Makefile

```diff
@@ -79,14 +80,12 @@ tidy: ## Update dependencies
 clean: ## Clean up the build artifacts
 	$(Q)rm -rf build
 
-lint-dev:  ## Run golangci-lint with all checks enabled (development purpose only)
-	./hack/tests/check-lint.sh dev
-
-lint-fix: ## Run golangci-lint automatically fix (development purpose only)
-	./hack/tests/check-lint.sh fix
-
-lint: ## Run golangci-lint with all checks enabled in the ci
-	./hack/tests/check-lint.sh ci
+lint: ## Install and run golangci-lint checks
+ifneq (${GOLANGCI_LINT_VER}, "$(shell ./bin/golangci-lint --version 2>/dev/null | cut -b 27-32)")
+	@echo "golangci-lint missing or not version '${GOLANGCI_LINT_VER}', downloading..."
+	curl -sSfL "https://raw.githubusercontent.com/golangci/golangci-lint/v${GOLANGCI_LINT_VER}/install.sh" | sh -s -- -b ./bin "v${GOLANGCI_LINT_VER}"
+endif
+	./bin/golangci-lint --timeout 5m run
```

Comment:
```suggestion
GOLANGCI_LINT_VER = "1.30.0"

lint: setup-lint ## Install and run golangci-lint checks
	$(LINT) --timeout 5m run

setup-lint: 
ifeq (, $(shell which golangci-lint --version 2 >/dev/null| cut -b 27-32))
	@{ \
	set -e ;\
	mkdir -p bin ;\
	echo "donwloading and installing golangci-lint@${GOLANGCI_LINT_VER} ..." ;\
	curl -sSfL "https://raw.githubusercontent.com/golangci/golangci-lint/v${GOLANGCI_LINT_VER}/install.sh" | sh -s -- -b ./bin "v${GOLANGCI_LINT_VER}" ;\
	}
LINT=$(realpath ./bin/golangci-lint)
else
LINT=$(shell which golangci-lint)
endif
```

Reply:
After check it better, I think it should be as the targets that we scaffolded with the tool.

Reply:
I don't understand... it only downloads if someone invokes `make lint` (in which case it is needed).  It only downloads if the version is the version defined, it doesn't overwrite the global GOBIN and it works.  What is current on master does not work.  IF there is some additional features or desire, perhaps we can create an issue with details and create a future PR? 
I really can't understand the comment as written or what is intended.

Reply:
Following a few motivations for the suggestion:
- Following the project standard. (See the Makefile of the projects generated by the tool)
- In your implementation, it will always download the lint in the bin/ dir. In the suggestion, it ONLY downloads the lint if it is NOT be installed already in the local env. See that it not necessary needs to be in the bin/ dir of the project.
- Allow works with the lint installed outside of the project. 
- Maintenance ability

Reply:
I am closing this one since we already spoke about. Please, let feel free to ping me when you are able to address the nit change.